src/main/java/duke/command/SearchCommand.java

```diff
@@ -0,0 +1,77 @@
+package duke.command;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.exceptions.IncompleteDukeCommandException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * The {@code SearchCommand} class represents a command that allows the user to search for tasks.
+ * The user is able to search for tasks either through their description or
+ * their date (if the type of task supports it).
+ */
+public class SearchCommand extends Command {
+
+    private String searchParameter;
+    private List<Task> results;
+
+    /**
+     * Constructs a {@code SearchCommand} with the specified {@code searchParameter}.
+     *
+     * @param searchParameter the string for which to search for.
+     */
+    public SearchCommand(String searchParameter) {
+        this.searchParameter = searchParameter;
+        results = new ArrayList<>();
```

Comment:
I do agree with you that standardising the use of `this` for results would look cleaner and better.
Although strangely the advanced style guide actually suggests otherwise. I suppose at this moment it is based on personal preference. 😆 

![Screenshot 2020-09-01 at 7 54 58 PM](https://user-images.githubusercontent.com/1428635/91847420-05cbaf00-ec8d-11ea-9b18-f5a2ea1d4b25.png)